torch/csrc/jit/tensorexpr/block_codegen.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+
+#include <ATen/ATen.h>
+#include <torch/csrc/jit/resource_guard.h>
+#include <torch/csrc/jit/tensorexpr/analysis.h>
+#include <torch/csrc/jit/tensorexpr/codegen.h>
+#include <torch/csrc/jit/tensorexpr/ir.h>
+#include <torch/csrc/jit/tensorexpr/ir_printer.h>
+#include <torch/csrc/jit/tensorexpr/ir_visitor.h>
+#include <torch/csrc/jit/tensorexpr/unique_name_manager.h>
+
+namespace torch {
+namespace jit {
+namespace tensorexpr {
+// A class that overrides the underlying IRPrinter to produce Block.
+class BlockPrinter : public IRPrinter {
+ public:
+  explicit BlockPrinter(std::ostream* os, const BlockAnalysis* block_analysis)
+      : IRPrinter(*os), block_analysis_(block_analysis) {}
+  void visit(const For* v) override;
```

Comment:
Minor: virtual functions could be marked as private whenever possible.

Reply:
Thanks for this! Done